provider/execution/execution.go

```diff
@@ -107,11 +107,14 @@ func (e TableExecutor) doMultiplexResolve(ctx context.Context, clients []schema.
 	logger := clients[0].Logger()
 	logger.Debug("multiplexing client", "count", len(clients), "table", e.Table.Name)
 	defer close(diagsChan)
+	numberOfClients := 0
 	for _, client := range clients {
 		// we can only limit on a granularity of a top table otherwise we can get deadlock
 		if err := e.goroutinesSem.Acquire(ctx, 1); err != nil {
-			return totalResources, allDiags.Add(ClassifyError(err, diag.WithResourceName(e.ResourceName)))
+			allDiags = allDiags.Add(ClassifyError(err, diag.WithResourceName(e.ResourceName)))
+			break
```

Comment:
you want to continue here, if you break you stop creating clients, unless this is intentional. 
if it is intentional I would add a log message that we stopped adding more clients and a better diagnostic that we reached MAX goroutines and details that you should increase them.